src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1863,12 +1863,13 @@ static int client_handle_message(sd_dhcp_client *client, DHCPMessage *message, i
 
                 client_initialize_io_events(client, client_receive_message_udp);
 
-                if (notify_event) {
+                if (IN_SET(client->state, DHCP_STATE_RENEWING, DHCP_STATE_REBINDING) &&
+                    notify_event == SD_DHCP_CLIENT_EVENT_IP_ACQUIRE)
+                        /* TODO: hmm, maybe this is a bug... */
+                        log_dhcp_client(client, "client_handle_ack() returns SD_DHCP_CLIENT_EVENT_IP_ACQUIRE during DHCP client is %s address, ignoring to trigger callback.",
```

Comment:
`"client_handle_ack() returned SD_DHCP_CLIENT_EVENT_IP_ACQUIRE while DHCP client was %s the address, skipping callback."`